assets/js/components/FileUpload.vue

```diff
@@ -55,15 +57,54 @@ export default {
   methods: {
     onSelect() {
       const file = this.$refs.file.files[0];
-      this.uploaded_files.push(file.name);
+      this.addFile(file.name);
+    },
+    addFile(file) {
+      axios
+        .post("/api/documents", {
+          file: file
+        })
+        .then(response => {
+          this.uploaded_files.push({
+            id: response.data.id,
+            fileName: response.data.originalFileName
+          });
+        });
     },
     removeFile(file) {
-      this.uploaded_files.splice(this.uploaded_files.indexOf(file), 1);
+      // if we're editing vs creating
+      axios
+        .delete("/api/conditions/" + this.conditionId + "/documents", {
```

Comment:
Question: Is there a way to use the Sails route functionality to do this? (i.e. if we change the URL, it's dynamically updated)

Reply:
good question... the route helper is available server-side on the sails object. This is not available on the client side, but I wonder if we could somehow make it available. Will investigate.

Reply:
I'm going to create a ticket for this and circle back around to it later. It's a good idea, just not sure how to make the route helper available on the client side, but I think it's possible.